.github/ubuntu_focal.yml

```diff
@@ -0,0 +1,54 @@
+name: Ubuntu-Focal
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+  schedule:
+    - cron: '0 5 * * *'
+
+jobs:
+  industrial_ci:
+    name: Ubuntu-Focal
+    runs-on: ubuntu-latest
+    env:
+      CI_NAME: Ubuntu-Focal
+      OS_NAME: ubuntu
+      OS_CODE_NAME: focal
+      ROS_DISTRO: noetic
+      ROS_REPO: main
+      ADDITIONAL_DEBS: 'clang-tidy'
+      UPSTREAM_WORKSPACE: 'dependencies.rosinstall'
+      ROSDEP_SKIP_KEYS: "iwyu ros_industrial_cmake_boilerplate"
+      CCACHE_DIR: "/home/runner/work/plugin_loader/plugin_loader/Ubuntu-Focal/.ccache"
+      BEFORE_RUN_TARGET_TEST_EMBED: "ici_with_unset_variables source /root/target_ws/install/setup.bash"
+      TARGET_CMAKE_ARGS: "-DCMAKE_BUILD_TYPE=Debug -DPLUGIN_LOADER_ENABLE_TESTING=ON -DPLUGIN_LOADER_ENABLE_CLANG_TIDY=ON"
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Free Disk Space
+        run: |
+          sudo swapoff -a
+          sudo rm -f /swapfile
+          sudo apt clean
+          docker rmi $(docker image ls -aq)
+          df -h
+
+      - name: Prepare ccache timestamp
+        id: ccache_cache_timestamp
+        shell: cmake -P {0}
+        run: |
+          string(TIMESTAMP current_date "%Y-%m-%d-%H;%M;%S" UTC)
+          message("::set-output name=timestamp::${current_date}")
+
+      - name: ccache cache files
+        uses: actions/cache@v1.1.0
+        with:
+          path: ${{ env.CI_NAME }}/.ccache
+          key: ${{ env.CI_NAME }}-ccache-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
+          restore-keys: |
+            ${{ env.CI_NAME }}-ccache-
+
+      - uses: 'ros-industrial/industrial_ci@master'
```

Comment:
Really doesn't feel like industrial_ci should be pulled in here with all the overhead of docker, ros packages, etc. With a rosdep step not needed a straightforward workspace build would be pretty simple.

Reply:
Agreed. I'll see what I can do. This won't run until I merge this though right?

Reply:
Looks like it won't though that seems odd since the workflow in the branch is the one that runs when you change something.